lib/ret_web/controllers/api/v1/oauth_controller.ex

```diff
@@ -29,7 +35,31 @@ defmodule RetWeb.Api.V1.OAuthController do
     end
   end
 
-  def show(conn, %{"type" => "discord", "error" => "access_denied", "state" => state}) do
+  # handle the chat app oauth information: discord, slack etc.
+  def show(conn, %{"type" => _type} = params) do
+    handle_chat_oauth(params, conn)
+    # handle_oauth(type, params, conn)
+  end
+
+  # def handle_oauth(type, params, conn) do
+  #   case params do
+  #     %{"error" => _} -> handle_chat_error(params, conn)
+  #     _ -> handle_chat_oauth(params, conn)
+  #   end
+  # end
+
+  # def handle_chat_error(params, conn) do
+  #   case
+
+
+  # def handle_oauth("slack", params, conn) do
+  #   case params do
+  #     %{"error" => _} -> handle_chat_error(params)
+  #     _ -> handle_chat_oauth(params)
+  #   end
+  # end
```

Comment:
I would just delete all this commented code. IMO, it's more distracting when reading this code than it is helpful and the code is straightforward enough to reproduce in the future if we need it.